src/transformStateWithClones.js

```diff
@@ -7,7 +7,45 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyState = {
+    ...state,
+  };
+
+  const history = [];
```

Comment:
About variables naming. That is a comment from one mentor:

This variable should be named in camelCase
UPPER_CASE -> constants(as above, lines 31-35)
camelCase -> calculated value

So, I think you need to use  this type of vars - UPPER_CASE.

Reply:
> About variables naming. That is a comment from one mentor:
> 
> This variable should be named in camelCase UPPER_CASE -> constants(as above, lines 31-35) camelCase -> calculated value
> 
> So, I think you need to use this type of vars - UPPER_CASE.

It's not constant in this case, add dynamically new values to it. Constants are variables that never changes